File: dotnet/src/xjustiz.core-dotnet/Models/Entities/Telekommunikation.cs
namespace xjustiz.core_dotnet.Models.Entities;

using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Helpers;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Informationen zur Telekommunikation (Telefon, Email, etc.).<br/>
/// <u><b>Telecommunication:</b></u> Information on telecommunication (phone, email, etc.).
/// </summary>
[XJustizAvailability(XJustizVersion.V2_1_0)]
[XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
public class Telekommunikation
{
    /// <summary>
    /// Art der Telekommunikation als CodeValue.<br/>
    /// <u><b>Telecommunication type:</b></u> Type of telecommunication as a code.
    /// </summary>
    [XmlElement("telekommunikationsart", Namespace = SchemeConstants.XJustiz_Tns)]
    [XJustizAvailability(XJustizVersion.V2_1_0)]
    [XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
    public Telekommunikationsart? Telekommunikationsart { get; set; }

    /// <summary>
    /// Hier wird die Telefonnummer, Faxnummer, E-Mail-Adresse, Safe-ID oder dergleichen mitgeteilt.<br/>
    /// <u><b>Connection:</b></u> The telephone number, fax number, e-mail address, Safe-ID, or similar is communicated here.
    /// </summary>
    [XmlElement("verbindung", Namespace = SchemeConstants.XJustiz_Tns)]
    [XJustizAvailability(XJustizVersion.V2_1_0)]
    [XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
    public string? Verbindung { get; set; }
}

File: java/src/main/java/de/xjustiz/core/models/Telekommunikation.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlElement;
import org.jetbrains.annotations.Nullable;

/**
 * Informationen zur Telekommunikation (Telefon, Email, etc.).
 * <p>
 *  <u><b>Telecommunication:</b></u> Information on telecommunication (phone, email, etc.).
 */
@XmlAccessorType(XmlAccessType.FIELD)
public class Telekommunikation {

    /**
     * Art der Telekommunikation als CodeValue.
     * <p>
     *  <u><b>Telecommunication type:</b></u> Type of telecommunication as a code.
     */
    @XmlElement(name = "telekommunikationsart", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "telekommunikationsart", namespace = "http://www.xjustiz.de")
    @JsonProperty("Telekommunikationsart")
    @Nullable
    private Telekommunikationsart telekommunikationsart;

    /**
     * Hier wird die Telefonnummer, Faxnummer, E-Mail-Adresse, Safe-ID oder dergleichen mitgeteilt.
     * <p>
     *  <u><b>Connection:</b></u> The telephone number, fax number, e-mail address, Safe-ID, or similar is communicated here.
     */
    @XmlElement(name = "verbindung", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "verbindung", namespace = "http://www.xjustiz.de")
    @JsonProperty("Verbindung")
    @Nullable
    private String verbindung;

    /**
     * Default constructor.
     */
    public Telekommunikation() {
    }

    /**
     * Art der Telekommunikation als CodeValue.
     *
     * @return the telekommunikationsart
     */
    @Nullable
    public Telekommunikationsart getTelekommunikationsart() {
        return telekommunikationsart;
    }

    /**
     * Sets the telekommunikationsart.
     *
     * @param telekommunikationsart the telekommunikationsart to set
     */
    public void setTelekommunikationsart(@Nullable Telekommunikationsart telekommunikationsart) {
        this.telekommunikationsart = telekommunikationsart;
    }

    /**
     * Hier wird die Telefonnummer, Faxnummer, E-Mail-Adresse, Safe-ID oder dergleichen mitgeteilt.
     *
     * @return the verbindung
     */
    @Nullable
    public String getVerbindung() {
        return verbindung;
    }

    /**
     * Sets the verbindung.
     *
     * @param verbindung the verbindung to set
     */
    public void setVerbindung(@Nullable String verbindung) {
        this.verbindung = verbindung;
    }

}
